File: src/Library/ObjectModel/LanguageSetupStep.vb
Public Class LanguageSetupStep
    Inherits Pixa.Soundbridge.Library.SetupStep

    Friend Sub New(ByVal client As ISoundbridgeClient)
        MyBase.New(client)
    End Sub

    Protected Overrides ReadOnly Property Name() As String
        Get
            Return "Language"
        End Get
    End Property

    Public Overrides Function GetSelectionList() As String()
        Return Client.ListLanguages
    End Function

    Public Overrides Sub MakeSelection(ByVal value As String)
        Dim r As String = Client.SetLanguage(value)

        If r = "OK" Then Exit Sub
        If r = "ParameterError" Then Throw New ArgumentException("value was not a valid language", "value")
        If r = "GenericError" Then Throw New ArgumentException(String.Format("Couldn't set language to {0}", value), "value")
    End Sub
End Class

File: src/Pixa.Soundbridge/LanguageSetupStep.cs
using Pixa.Soundbridge.Client;
using System;

namespace Pixa.Soundbridge
{
    public class LanguageSetupStep : SetupStep
    {
        internal LanguageSetupStep(ISoundbridgeClient client) : base(client)
        {
        }

        protected override string Name
        {
            get
            {
                return "Language";
            }
        }

        public override string[] GetSelectionList()
        {
            return Client.ListLanguages();
        }

        public override void MakeSelection(string value)
        {
            string r = Client.SetLanguage(value);
            if (r == "OK")
                return;
            if (r == "ParameterError")
                throw new ArgumentException("value was not a valid language", "value");
            if (r == "GenericError")
                throw new ArgumentException(string.Format("Couldn't set language to {0}", value), "value");
        }
    }
}